src/back/msl.rs

```diff
@@ -451,7 +451,11 @@ impl<W: Write> Writer<W> {
                         write!(self.out, ".{}", COMPONENTS[index as usize])?;
                         Ok(MaybeOwned::Owned(crate::TypeInner::Scalar { kind, width }))
                     }
-                    crate::TypeInner::Array { base, size } => {
+                    crate::TypeInner::Array {
+                        base,
+                        size,
+                        stride: _,
```

Comment:
Does Metal have a stride or is it just that it is not planned yet to support strides for our Metal back-end?

Reply:
we'll need to insert padding ourselves, that's a big TODO for now